libcontainer/cgroups/fscommon/rdma.go

```diff
@@ -0,0 +1,121 @@
+// +build linux
+
+package fscommon
+
+import (
+	"bufio"
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"golang.org/x/sys/unix"
+)
+
+// parseRdmaKV: parses raw string to RdmaEntry.
+func parseRdmaKV(raw string, entry *cgroups.RdmaEntry) error {
+	var value uint64
+	var err error
+
+	parts := strings.SplitN(raw, "=", 3)
+
+	if len(parts) == 2 {
+		if parts[1] == "max" {
+			value = math.MaxUint64
+		} else {
+			value, err = strconv.ParseUint(parts[1], 10, 32)
+			if err != nil {
+				return err
+			}
+		}
+		if parts[0] == "hca_handle" {
+			entry.HcaHandles = uint32(value)
+		} else if parts[0] == "hca_object" {
+			entry.HcaObjects = uint32(value)
+		}
+	} else {
+		return errors.New("Unable to parse RDMA entry")
+	}
+
+	return nil
+}
+
+// readRdmaEntries: Reads and converts array of rawstrings to RdmaEntries from file.
+func readRdmaEntries(dir, file string) ([]cgroups.RdmaEntry, error) {
+	rdmaEntries := make([]cgroups.RdmaEntry, 0)
+	fd, err := cgroups.OpenFile(dir, file, unix.O_RDONLY)
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close() //nolint:errorlint
+	scanner := bufio.NewScanner(fd)
+	for scanner.Scan() {
+		strEntry := scanner.Text()
+		parts := strings.SplitN(strEntry, " ", 4)
```

Comment:
SplitN 3 here?

Reply:
See above; doing so wouldn't allow detecting len(parts) > 3

Reply:
Oh, Good. Thanks!

Reply:
@fuweid @thaJeztah Can i mark this as resolved ? As SplitN 3 is needed here.

Reply:
Yes, please. :)